cmake/SetupSpheral.cmake

```diff
@@ -83,8 +82,6 @@ if(ENABLE_CUDA)
   list(APPEND SPHERAL_CXX_DEPENDS cuda)
 endif()
 
-option(BOOST_HEADER_ONLY "only use the header only components of Boost" OFF)
```

Comment:
We use the header-only boost option, is it being removed because it's the default?

Reply:
It was thought to be unnecessary since we no longer build boost with cmake. Does it actually make a difference for your builds?

Reply:
Were hoping to do a separate PR soon where we can update TPLs to more recent versions. This PR only updates TPLs in order to make the latest version of spack happy.